app/controllers/markets_controller.rb

```diff
@@ -1,7 +1,22 @@
 class MarketsController < ApplicationController
+  def search
+  end
+
   def index
+    if params[:use_my_location] != "0"
+      location = Geocoder.search(params[:use_my_location])
+    end
+
+    location = Geocoder.search(params[:search])
```

Comment:
High potential for refactor into services, facades and poros for maintainability in the future.